modules/dnn/src/dnn.cpp

```diff
@@ -239,11 +239,10 @@ class BackendRegistry
 #endif
 
 #ifdef HAVE_CUDA
-        if (haveCUDA() && cuda4dnn::isDeviceCompatible())
+        if (haveCUDA())
         {
             backends.push_back(std::make_pair(DNN_BACKEND_CUDA, DNN_TARGET_CUDA));
-            if (cuda4dnn::doesDeviceSupportFP16())
-                backends.push_back(std::make_pair(DNN_BACKEND_CUDA, DNN_TARGET_CUDA_FP16));
+            backends.push_back(std::make_pair(DNN_BACKEND_CUDA, DNN_TARGET_CUDA_FP16));
```

Comment:
This might be a breaking change. The issue is that `BackendRegistry` is a singleton object but the available targets for CUDA depend on the current device that has been selected.

If the user initially used a device without FP16 support and then switches to a device with FP16 support, FP16 target won't be returned with `getAvailableTargets` since the registry is initialized when the device without FP16 support was present.